curdleproofs/curdleproofs/util.py

```diff
@@ -1,74 +1,71 @@
 from random import randint
 from math import log2
-from typing import List, Tuple, Type, TypeVar, Union
-from py_ecc.typing import (
-    Optimized_Field,
-    Optimized_Point2D,
-    Optimized_Point3D,
-    FQ as FQ_type,
-)
-from py_ecc.optimized_bls12_381.optimized_curve import (
-    curve_order,
-    G1,
-    multiply,
-    normalize,
-    FQ,
-)
-from py_ecc.bls.hash import os2ip
-from py_ecc.bls.g2_primitives import pubkey_to_G1
-from eth_typing import BLSPubkey
-from py_ecc.bls.point_compression import compress_G1
+from typing import List, TypeVar, NewType
+from py_arkworks_bls12381 import G1Point, Scalar
 
 
-class Fr(FQ_type):
-    field_modulus: int = curve_order
+CURVE_ORDER = 52435875175126190479447740508185965837690552500527637822603658699938581184513
+# Generator
+G1 = G1Point()
+# Point at infinity over FQ
+Z1 = G1Point.identity()
 
 
-PointAffine = Optimized_Point2D[Optimized_Field]
-PointProjective = Optimized_Point3D[Optimized_Field]
+BLSPubkey = NewType('BLSPubkey', bytes)  # bytes48
 
 
-def point_affine_to_bytes(point: PointAffine) -> bytes:
-    return point[0].n.to_bytes(48, "big") + point[1].n.to_bytes(48, "big")
+def g1_is_inf(point: G1Point):
+    return point == G1
```

Comment:
Hmm, this seems wrong? Shouldn't this be checking for the point at infinity? Maybe that's `Z1`?

With these changes it seems to check whether `point` is the generator `G1`.

Reply:
Yeah good catch

Reply:
Oh tests pass even with the `Z1` change? That's cool.

Reply:
Not sure if cool haha maybe that path is not excercised enough?